make cross broken for Windows on master

**Description**

Unable to build `dockerd` for `windows` using the `make cross` command on the master branch
Note - This issue is not seen on the 19.03 branch


<!--
Briefly describe the problem you are having in a few paragraphs.
-->

**Steps to reproduce the issue:**
1. `git checkout master`
2. `make DOCKER_CROSSPLATFORMS=windows/amd64 cross`

**Describe the results you received:**

```
 🐳 ~/go-workspace/src/github.com/moby$ make DOCKER_CROSSPLATFORMS=windows/amd64 cross
docker build  --build-arg=GO_VERSION  -f "Dockerfile" --build-arg CROSS=true --build-arg DOCKER_CROSSPLATFORMS --output=bundles/ --target=cross --build-arg VERSION --build-arg DOCKER_GITCOMMIT --build-arg PRODUCT --build-arg PLATFORM --build-arg DEFAULT_PRODUCT_LICENSE .
[+] Building 2.3s (39/40)                                                                                                                                                                                           
 => [internal] load .dockerignore                                                                                                                                                                              0.0s
 => => transferring context: 34B                                                                                                                                                                               0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                           0.0s
 => => transferring dockerfile: 37B                                                                                                                                                                            0.0s
 => resolve image config for docker.io/docker/dockerfile:1.1.3-experimental                                                                                                                                    1.0s
 => CACHED docker-image://docker.io/docker/dockerfile:1.1.3-experimental@sha256:888f21826273409b5ef5ff9ceb90c64a8f8ec7760da30d1ffbe6c3e2d323a7bd                                                               0.0s
 => [internal] load metadata for docker.io/library/golang:1.13.6-buster                                                                                                                                        0.4s
 => [internal] load build context                                                                                                                                                                              0.3s
 => => transferring context: 450.30kB                                                                                                                                                                          0.3s
 => [base 1/3] FROM docker.io/library/golang:1.13.6-buster@sha256:f6cefbdd25f9a66ec7dcef1ee5deb417882b9db9629a724af8a332fe54e3f7b3                                                                             0.0s
 => CACHED [base 2/3] RUN echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache                                                                                           0.0s
 => CACHED [base 3/3] RUN sed -ri "s/(httpredir|deb).debian.org/${APT_MIRROR:-deb.debian.org}/g" /etc/apt/sources.list  && sed -ri "s/(security).debian.org/${APT_MIRROR:-security.debian.org}/g" /etc/apt/so  0.0s
 => CACHED [cross-true 1/4] RUN dpkg --add-architecture arm64                                                                                                                                                  0.0s
 => CACHED [cross-true 2/4] RUN dpkg --add-architecture armel                                                                                                                                                  0.0s
 => CACHED [cross-true 3/4] RUN dpkg --add-architecture armhf                                                                                                                                                  0.0s
 => CACHED [cross-true 4/4] RUN --mount=type=cache,sharing=locked,id=moby-cross-true-aptlib,target=/var/lib/apt     --mount=type=cache,sharing=locked,id=moby-cross-true-aptcache,target=/var/cache/apt        0.0s
 => CACHED [runtime-dev-cross-false 1/1] RUN --mount=type=cache,sharing=locked,id=moby-cross-false-aptlib,target=/var/lib/apt     --mount=type=cache,sharing=locked,id=moby-cross-false-aptcache,target=/var/  0.0s
 => CACHED [runtime-dev-cross-true 1/1] RUN --mount=type=cache,sharing=locked,id=moby-cross-true-aptlib,target=/var/lib/apt     --mount=type=cache,sharing=locked,id=moby-cross-true-aptcache,target=/var/cac  0.0s
 => CACHED [tini 1/4] RUN --mount=type=cache,sharing=locked,id=moby-tini-aptlib,target=/var/lib/apt     --mount=type=cache,sharing=locked,id=moby-tini-aptcache,target=/var/cache/apt         apt-get update   0.0s
 => CACHED [tini 2/4] COPY hack/dockerfile/install/install.sh ./install.sh                                                                                                                                     0.0s
 => CACHED [tini 3/4] COPY hack/dockerfile/install/tini.installer ./                                                                                                                                           0.0s
 => CACHED [tini 4/4] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod         PREFIX=/build ./install.sh tini                                                    0.0s
 => CACHED [binary-base 1/6] COPY --from=tini        /build/ /usr/local/bin/                                                                                                                                   0.0s
 => CACHED [runc 1/3] COPY hack/dockerfile/install/install.sh ./install.sh                                                                                                                                     0.0s
 => CACHED [runc 2/3] COPY hack/dockerfile/install/runc.installer ./                                                                                                                                           0.0s
 => CACHED [runc 3/3] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod         PREFIX=/build ./install.sh runc                                                    0.0s
 => CACHED [binary-base 2/6] COPY --from=runc        /build/ /usr/local/bin/                                                                                                                                   0.0s
 => CACHED [containerd 1/4] RUN --mount=type=cache,sharing=locked,id=moby-containerd-aptlib,target=/var/lib/apt     --mount=type=cache,sharing=locked,id=moby-containerd-aptcache,target=/var/cache/apt        0.0s
 => CACHED [containerd 2/4] COPY hack/dockerfile/install/install.sh ./install.sh                                                                                                                               0.0s
 => CACHED [containerd 3/4] COPY hack/dockerfile/install/containerd.installer ./                                                                                                                               0.0s
 => CACHED [containerd 4/4] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod         PREFIX=/build ./install.sh containerd                                        0.0s
 => CACHED [binary-base 3/6] COPY --from=containerd  /build/ /usr/local/bin/                                                                                                                                   0.0s
 => CACHED [proxy 1/3] COPY hack/dockerfile/install/install.sh ./install.sh                                                                                                                                    0.0s
 => CACHED [rootlesskit 2/4] COPY hack/dockerfile/install/rootlesskit.installer ./                                                                                                                             0.0s
 => CACHED [rootlesskit 3/4] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod         PREFIX=/build/ ./install.sh rootlesskit                                     0.0s
 => CACHED [rootlesskit 4/4] COPY ./contrib/dockerd-rootless.sh /build                                                                                                                                         0.0s
 => CACHED [binary-base 4/6] COPY --from=rootlesskit /build/ /usr/local/bin/                                                                                                                                   0.0s
 => CACHED [proxy 2/3] COPY hack/dockerfile/install/proxy.installer ./                                                                                                                                         0.0s
 => CACHED [proxy 3/3] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod         PREFIX=/build ./install.sh proxy                                                  0.0s
 => CACHED [binary-base 5/6] COPY --from=proxy       /build/ /usr/local/bin/                                                                                                                                   0.0s
 => CACHED [binary-base 6/6] WORKDIR /go/src/github.com/docker/docker                                                                                                                                          0.0s
 => ERROR [build-cross 1/1] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=bind,target=/go/src/github.com/docker/docker         hack/make.sh cross                                       0.3s
------                                                                                                                                                                                                              
 > [build-cross 1/1] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=bind,target=/go/src/github.com/docker/docker         hack/make.sh cross:                                                  
#39 0.181 
#39 0.235 Removing /build/bundles/
#39 0.236 
#39 0.242 ---> Making bundle: cross (in /build/bundles/cross)
#39 0.243 Cross building: /build/bundles/cross/windows/amd64
#39 0.256 mkdir: cannot create directory 'autogen': Read-only file system
#39 0.256 mkdir: cannot create directory 'autogen': Read-only file system
------
failed to solve with frontend dockerfile.v0: failed to solve with frontend gateway.v0: rpc error: code = Unknown desc = failed to build LLB: executor failed running [/bin/sh -c hack/make.sh cross]: runc did not terminate sucessfully
make: *** [cross] Error 1
```

**Describe the results you expected:**
Expect no errors

--- a/Dockerfile
+++ b/Dockerfile
@@ -354,6 +354,7 @@ FROM binary-base AS build-cross
 ARG DOCKER_CROSSPLATFORMS
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=bind,target=/go/src/github.com/docker/docker \
+    --mount=type=tmpfs,target=/go/src/github.com/docker/docker/autogen \
         hack/make.sh cross
 
 FROM scratch AS binary

--- a/Makefile
+++ b/Makefile
@@ -172,7 +172,14 @@ cross: ## cross build the binaries for darwin, freebsd and\nwindows
 
 cross: BUILD_OPTS += --build-arg CROSS=true --build-arg DOCKER_CROSSPLATFORMS
 
-binary dynbinary cross: buildx
+# This is only used to work around read-only bind mounts of the source code into
+# binary build targets. We end up mounting a tmpfs over autogen which allows us
+# to write build-time generated assets even though the source is mounted read-only
+# ...But in order to do so, this dir needs to already exist.
+autogen:
+	mkdir -p autogen
+
+binary dynbinary cross: buildx autogen
 	$(BUILD_CMD) $(BUILD_OPTS) --output=bundles/ --target=$@ $(VERSION_AUTOGEN_ARGS) .
 
 build: target = --target=final

--- a/hack/make/.go-autogen
+++ b/hack/make/.go-autogen
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-rm -rf autogen
+rm -rf autogen/*
 
 source hack/dockerfile/install/runc.installer
 source hack/dockerfile/install/tini.installer

